code/game/objects/effects/anomalies/anomaly_hallucination.dm

```diff
@@ -37,8 +37,7 @@
 
 		// Everyone else
 		var/dist = sqrt(1 / max(1, get_dist(near, location)))
-		near.hallucination += strength * dist
-		near.hallucination = clamp(near.hallucination, 0, 150)
+		near.hallucination = ADDCLAMP(near.hallucination, strength * dist, 0, 150)
```

Comment:
This case is why I wanted to keep it as proc. If a given syntax is complex, you'll do that multiple times.

Reply:
can you declare a temp var inside the define?

Reply:
also, a sum is probably a much cheaper operation than a proc call, even if done three times.

Reply:
> can you declare a temp var inside the define?

You can't. Unless DM supports an arrow function. It's just not possible logic afaik.

Reply:
> also, a sum is probably a much cheaper operation than a proc call, even if done three times.

I don't get why you say sum here.

Reply:
> can you declare a temp var inside the define?

You can use global/static variables to temporarilly store values. See my PR on virtual multi-z